actor/src/main/resources/reference.conf

```diff
@@ -1144,6 +1144,12 @@ pekko {
         # Defaults to a system dependent lookup (on Unix like OSes, will attempt to parse /etc/resolv.conf, on
         # other platforms, will default to 1).
         ndots = default
+
+        # The policy used to generate dns transaction ids. Options are sequence, thread-local-random or secure-random.
+        # Defaults to thread-local-random similar to Netty, secure-random produces FIPS compliant random numbers but
+        # could block looking for entropy (these are short integers so are easy to bruit-force), sequence is the old
+        # behavior.
+        id-generator-policy = thread-local-random
```

Comment:
This needs to be `SecureRandom` by default or we have to use @Claudenw version as bare minimum. We should not have insecure algorithm by default

Reply:
If we are worried about contention on the secure random, we could add another id generator that uses 'ThreadLocal<SecureRandom>' (SecureRandom has some synchronization). It would be good to have a plain SecureRandom too because ThreadLocals can be problematic if you have lots of short lived threads.

Reply:
@Claudenw this PR handles duplicates

That said, we could add a generator based on what you were working on as an optional generator.

Reply:
It could be, but it comes with a lot of downsides. Its very easy to run out of entropy when a container starts which will block the startup of your application trying to generate the transactions id for DNS queries to join the cluster.

The other thing to note is that we are generating a short, it is trivial to brute-force 32k numbers if an attacker wanted to break DNS SecureRandom would not prevent that.

IMO the only reason to enable SecureRandom would be if you require it for FIPS.

Reply:
https://www.digitalocean.com/community/tutorials/how-to-setup-additional-entropy-for-cloud-servers-using-haveged is a way to resolve low entropy.

Reply:
So I just spoke with @Claudenw and his solution at https://github.com/Claudenw/IdSupplier/blob/main/src/org/example/ShortSupplier.java should do a better job at avoiding so many collision's due to 32k numbers, in other words it actually goes out if its way to minimize the amount of collisions detected by using enhanced double hashing + 2 buffers to also handle roll over

> IMO the only reason to enable SecureRandom would be if you require it for FIPS.

We have a person in our company that specializes in security who's on IETF and his recommendation was to use `SecureRandom` as a seed + a PRF such as SHA-256/HMAC. His view is that `SecureRandom` on every is excessive, however non crypto secure hashes is way too easy.

The common/trivial way to solve this issue appears to be using `SecureRandom` every time and check for collisions against in flight transactions, thats what systemd-resolved does for example.